### ISSUE TITLE
lambda_dssim equals to 0.8 is causing a massive grows in the number of splats

### ISSUE THREAD

**Issue by francescodisario** [nus-vv-streams/lapis-gs#6]
Hi,
First of all, great work! I'm very interested in your project. However, I noticed that when running the provided code on my machine, the number of Gaussians grows exponentially due to increasing lambda_dssim.

For example, in the bonsai scene, the number of Gaussians at resolution 1x exceeds 9 million (expected ~1.2 million). Resetting the hyperparameter to its default value brings the results back to a reasonable range, similar to vanilla 3DGS.

I'm wondering if this behavior is an issue with the code or something specific to my setup.

**Additional Details**
Scene: _bonsai_
Gaussian Count at Different Resolutions:

- res8: 2,029,086 (which is already quite high)
- res4: 4,532,349
- res2: 6,996,570
- res1: 9,207,378

With more complex scenes, this also leads to a CUDA out-of-memory error.

Any insights would be greatly appreciated!

**Comment by Shiyuang-scu** [comment]
Hey, thank you for your interest!

TL;DR, I'll say it is the main issue of current working flow of LapisGS, but addressing this problem is orthogonal to our work.

Through experiments, we found that the default loss function is not sensitive to low-resolution images, making optimization and desification failed. It is because in the default loss function, L1 loss is attached much more importance (0.8), but L1 loss is not sensitive to finer details, blurriness, or low-resolution artifacts. Therefore, **the loss, computed from the default loss function, would be small at low layers, disabling the parameter update and adaptive control for the low-layer Gaussian splats**.

There are lots of works try to solve this problem, such as improving the loss function or the densification criteria. We also believe incorporating these works in our framework would produce a better layered model. However, in my opinion, we need to notice that forcing the optimization and adaptive control to pay attention to finer details, blurriness, or low-resolution artifacts is always at the cost of model size. In short, **we always have to trade the storage efficiency for visual quality**... just at different degrees. 

Nevertheless, we admit this issue and are trying to improve it with GS compression, by incorporating our previous work [1]. 

Hope this reply could solve your question!

[1] Sketch and Patch: Efficient 3D Gaussian Representation for Man-Made Scenes. https://arxiv.org/abs/2501.13045
